src/Services/Exporter/WishlistToPdfExporter.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+/*
+ * This file was created by developers working at BitBag
+ * Do you need more information about us and what we do? Visit our https://bitbag.io website!
+ * We are hiring developers from all over the world. Join us and start your new, exciting adventure and become part of us: https://bitbag.io/career
+*/
+
+declare(strict_types=1);
+
+namespace BitBag\SyliusWishlistPlugin\Services\Exporter;
+
+use BitBag\SyliusWishlistPlugin\Command\Wishlist\WishlistItemInterface;
+use BitBag\SyliusWishlistPlugin\Exception\NoProductSelectedException;
+use BitBag\SyliusWishlistPlugin\Exception\ProductVariantNotFoundException;
+use BitBag\SyliusWishlistPlugin\Model\VariantPdfModelInterface;
+use BitBag\SyliusWishlistPlugin\Services\Generator\ModelCreator;
+use Doctrine\Common\Collections\Collection;
+use Dompdf\Dompdf;
+use Dompdf\Options;
+use Sylius\Component\Core\Model\ProductVariant;
+use Sylius\Component\Core\Repository\ProductVariantRepositoryInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Twig\Environment;
+
+final class WishlistToPdfExporter implements WishlistToPdfExporterInterface
+{
+    private ProductVariantRepositoryInterface $productVariantRepository;
+
+    private Environment $twigEnvironment;
+
+    private ModelCreator $modelCreator;
+
+    public function __construct(
+        ProductVariantRepositoryInterface $productVariantRepository,
+        Environment $twigEnvironment,
+        ModelCreator $modelCreator
+    ) {
+        $this->productVariantRepository = $productVariantRepository;
+        $this->twigEnvironment = $twigEnvironment;
+        $this->modelCreator = $modelCreator;
+    }
+
+    public function createModelToPdfAndExportToPdf(Collection $wishlistProducts, Request $request): void
+    {
+        $productsToExport = $this->createVariantModelToPdf($wishlistProducts, $request);
+
+        if (empty($productsToExport)) {
+            throw new NoProductSelectedException();
+        }
+        $this->exportToPdf($productsToExport);
+    }
+
+    private function createVariantModelToPdf(Collection $wishlistProducts, Request $request): array
+    {
+        /** @var WishlistItemInterface $wishlistProduct */
+        foreach ($wishlistProducts as $wishlistProduct) {
+            $selectedProducts[] = $this->createCollectionOfWishlistItems($wishlistProduct, $request);
+        }
+
+        return $selectedProducts;
```

Comment:
Do you filter null items somewhere?

Reply:
Yes,  function wishlistThrowException is responsible for it